lib/validation.js

```diff
@@ -97,5 +96,14 @@ function isValidLogger (logger) {
   return result
 }
 
-module.exports = { build, validate, serialize, isValidLogger }
+function schemaCompiler (schema) {
+  const validateFuncion = ajv.compile(schema)
+  return function (body) {
```

Comment:
Why are we creating this closure?
Is there some performance gain?
If not, could you just return `validateFuncion` (there is a typo, you are missing a 't' ;) ) and call `inputSchemaError` as were before?

Reply:
Bad news guys:
Currently master on my PC serve 63k in 10s meanwhile this branch only 58k.
I've tested 4xx response.

Unfortunately, `ajv` stores the error always on the function and don't return the error directly.
This closure is made to fix an interface for the validation function.

@delvedor can you help me to find a way to remove this closure?

Reply:
I think you should keep `validate` as before, and update only `schemaCompiler`.
I've created a [gist](https://gist.github.com/delvedor/eea2097e096ee28d7af6f0351f7724b8) :)

Reply:
In that way the interface is a little bit complicated for other validation library like `Joi`.
```
const error = schema.validate(body)
```
That interface is very common.

But, if there're not other solutions that allow us to keep the performance, the `ajv` interface is fine.

Reply:
> In that way the interface is a little bit complicated for other validation library like Joi.

Ok, now I understood better the issue. I'll think about it :)